dask/delayed.py

```diff
@@ -208,11 +209,15 @@ def delayed(obj, name=None, pure=False):
     """
     if isinstance(obj, Delayed):
         return obj
+
     task, dasks = to_task_dasks(obj)
-    root = not dasks
-    name = name or '%s-%s' % (type(obj).__name__, tokenize(task, pure=True))
-    dasks.append({name: task})
-    return Delayed(name, dasks, pure=pure, root=root)
+
+    if not dasks and callable(obj):
+        return DelayedFunction(obj, pure=pure, prefix=prefix)
```

Comment:
Perhaps, but there's still no harm in attaching them. I frequently do `delayed = delayed(pure=True)` at the top of a file. With this definition, `delayed(1)` should still be valid imo, so forbidding pure with non-callables isn't the best (and you don't do that here). I just find it easier to reason about if you can say "`delayed` on things with dask graphs returns `Delayed`, otherwise `DelayedRoot`".